src/meta/cycleData/link/link.ts

```diff
@@ -0,0 +1,37 @@
+import { CountryIso } from 'meta/area'
+
+export type LinkLocation = {
+  column: string
+  entityProps: Record<string, string>
+  id: number
+  table: string
+}
+
+export enum LinkValidationStatusCode {
+  empty = 'empty',
+  enotfound = 'enotfound',
+  success = 'success',
+  urlParsing = 'urlParsing',
+}
+
+export type LinkProps = {
+  approved?: boolean
+  name: string
+  visit: {
+    code: LinkValidationStatusCode
+  }
+  visits: Array<{
+    code: LinkValidationStatusCode
+    timestamp: string
+  }>
+}
+
+export type Link = {
+  readonly id: number
+  readonly uuid: string
+  countryIso: CountryIso
+  link: string
+  location: LinkLocation
+  props: LinkProps
+  target: string
```

Comment:
We had it in our first notes but I agree we don't need it 👍

Reply:
@minotogna @sorja 
I am writing the query to upsert a link. It goes like:
```
insert into ${schemaCycle}.link 
         (country_iso, link, location, props, target)
         values ($1, $2, $3::jsonb, $4::jsonb, $5)
  on conflict (link, country, ???) do update
  props = ...
```
If the same link (url) exists in different locations, do we want to treat it as if it was actually two different links?
If so, we would need to have the 'location_id' as a new column to add a (link, country, location_id) unique constraint. 
Alternatively, we can have a column `locations` array instead of `location`. If the same url is in different places, then we just append the location to the existing link.

Reply:
I have an idea, let me know what you think:
We make each link/country unique, and we handle different locations with an array of link locations.
The hard part of having an array of locations is to update them when a link is moved/deleted. What we can do there is: when we run the validation for all links, first we set all link locations to null. Then we run the validation, which will add the updated link locations. At the end, any link that still has a null location can be deleted because it means there are no instances of it anymore.

Reply:
Some examples of links that are used multiple times in different places:
![image](https://github.com/openforis/fra-platform/assets/41337901/d21a062e-fc4e-4762-844f-9151384e808c)

Reply:
It will require a bit more pre-processing before inserting, but I think this is the way to go 👍